src/blocks/HighlightBlock/Component.tsx

```diff
@@ -0,0 +1,92 @@
+import { Article, Category, HighlightBlock as HighlightBlockProps } from '@/payload-types'
+import React from 'react'
+import { getPayload } from 'payload'
+import config from '@payload-config'
+import { HighlightedElement, HighlightElementsComponent } from './elements/Component'
+
+type Props = HighlightBlockProps
+
+const getProductElements = async (productArray: any[]): Promise<HighlightedElement[]> => {
+  const ids = productArray
+    ?.map((product) => {
+      if (product === null || product === undefined) return null
+      return typeof product === 'number' ? product : product.product
+    })
+    .filter((el) => typeof el === 'number')
+  const payload = await getPayload({ config })
+  const findResult = await payload.find({
+    collection: 'eu-products',
+    where: {
+      id: {
+        in: ids,
+      },
+    },
+  })
+
+  return findResult.docs
+    .filter((el) => el !== null && typeof el === 'object')
+    .map((el) => ({
+      name: el.name,
+      image: typeof el.logo === 'object' && el.logo?.url ? el.logo?.url : undefined,
+      link: el.link,
+      summary: el.description ?? undefined,
+    }))
+}
+
+const getCategoryElements = (categoriesArray: Category[]): HighlightedElement[] => {
+  return categoriesArray.map((el) => ({
+    name: el.name,
+    image: undefined,
+    summary: 'Category description',
+    link: 'https://www.google.com',
```

Comment:
```suggestion
      link: el.slug ?? '',
```
Can we use slug here?